inc/performance_optimizations/namespace.php

```diff
@@ -17,7 +15,14 @@ function bootstrap() {
  *
  */
 function increase_set_time_limit_on_async_upload() {
-	if ( ini_get( 'max_execution_time' ) < 120 ) {
-		set_time_limit( 120 );
+	if ( ! isset( $_SERVER['REQUEST_URI'] ) ) {
+		return;
+	}
+	if ( strpos( $_SERVER['REQUEST_URI'], '/wp-admin/async-upload.php' ) === false ) {
+		return;
+	}
+	if ( ini_get( 'max_execution_time' ) >= 120 ) {
```

Comment:
Oh actually, what if the max_execution_time is set to unlimited (`0`)?

Reply:
```suggest
	$max = ini_get( 'max_execution_time' );
	 if ( $max === 0 || $max >= 120 ) {
```